policy/modules/services/xserver.if

```diff
@@ -185,42 +185,6 @@ interface(`xserver_role',`
 	xserver_read_xkb_libs($2)
 
 	optional_policy(`
-		xdg_manage_all_cache($2)
-		xdg_relabel_all_cache($2)
-		xdg_watch_all_cache_dirs($2)
-		xdg_manage_all_config($2)
-		xdg_relabel_all_config($2)
-		xdg_watch_all_config_dirs($2)
-		xdg_manage_all_data($2)
-		xdg_relabel_all_data($2)
-		xdg_watch_all_data_dirs($2)
-
-		xdg_generic_user_home_dir_filetrans_cache($2, dir, ".cache")
-		xdg_generic_user_home_dir_filetrans_config($2, dir, ".config")
-		xdg_generic_user_home_dir_filetrans_data($2, dir, ".local")
-
-		xdg_generic_user_home_dir_filetrans_documents($2, dir, "Documents")
-		xdg_generic_user_home_dir_filetrans_downloads($2, dir, "Downloads")
-		xdg_generic_user_home_dir_filetrans_music($2, dir, "Music")
-		xdg_generic_user_home_dir_filetrans_pictures($2, dir, "Pictures")
-		xdg_generic_user_home_dir_filetrans_videos($2, dir, "Videos")
-
-		xdg_manage_documents($2)
-		xdg_relabel_documents($2)
-		xdg_watch_documents_dirs($2)
-		xdg_manage_downloads($2)
-		xdg_relabel_downloads($2)
-		xdg_watch_downloads_dirs($2)
-		xdg_manage_music($2)
-		xdg_relabel_music($2)
-		xdg_watch_music_dirs($2)
-		xdg_manage_pictures($2)
-		xdg_relabel_pictures($2)
-		xdg_watch_pictures_dirs($2)
-		xdg_manage_videos($2)
-		xdg_relabel_videos($2)
-		xdg_watch_videos_dirs($2)
-
 		xdg_cache_filetrans($2, mesa_shader_cache_t, dir, "mesa_shader_cache")
```

Comment:
Why not move this too?

Reply:
> I'm ok with this move, but instead of directly putting them in `userdom_unpriv_user_template()`, they should go in a new template called `userdom_xdg_user_template()` so that if a policy writer wants to add the XDG usage onto a new user they're creating, it is 1 line rather than 34.

My rationale for this move is that I ran into an issue with this on an SELinux-enabled headless server. This server does not have a display server (X, wayland) installed at all and the policy is built modularly. As a result the xdg.pp module is never built into the running policy. Because of this, the types for `xdg_config_t` and so on exist, but their `xdg_config_type` attributes are never set. The noticeable impact is that as a user, I do not have permission to access any objects with these types, to include `.config` and `.local` directories in my own home.

Reply:
I don't understand how my suggestion undermines your usage? `userdom_unpriv_user_template()` would call `userdom_xdg_user_template()`.

Reply:
> `userdom_unpriv_user_template()` would call `userdom_xdg_user_template()`.

I'm sorry I think I get it now. To resolve the issue with the `mesa_shader_cache_t` type, do you think that can just be wrapped in an `optional_policy` in that template?

Reply:
I think the mesa line should remain where it is.

Reply:
I have moved the xdg rules to a new `userdom_xdg_user_template()`